CodeGen/Generators/QuantityRelationsParser.cs

```diff
@@ -59,6 +59,24 @@ public static void ParseAndApplyRelations(string rootDir, Quantity[] quantities)
                 })
                 .ToList());
 
+            // We can infer division relations from multiplication relations.
+            relations.AddRange(relations
+                .Where(r => r.Operator is "*")
+                .Select(r => r with
+                {
+                    Operator = "/",
+                    LeftQuantity = r.ResultQuantity,
+                    LeftUnit = r.ResultUnit,
+                    ResultQuantity = r.LeftQuantity,
+                    ResultUnit = r.LeftUnit,
+                })
+                // Skip division between equal quantities because the ratio is already generated as part of the Arithmetic Operators.
+                .Where(r => r.LeftQuantity != r.RightQuantity)
+                .ToList());
+            
+            // Remove inferred relation "MassConcentration = Mass / Volume" because it duplicates "Density = Mass / Volume"
+            relations.RemoveAll(r => r is { Operator: "/", ResultQuantity.Name: "MassConcentration", LeftQuantity.Name: "Mass", RightQuantity.Name: "Volume" });
```

Comment:
Maybe we can solve this in a way that makes it more discoverable by moving this into the JSON file, how about this:

### Option 1) Skip division if multiple competing definitions

If there more than one inferred division definition `QuantityA / QuantityB` with different resulting quantity type, then skip all of them. They will have to be manually defined in the JSON file.

### Option 2) Allow defining opt-out in the JSON file

We could move the array to a structure like this:
```json
{
  "skipInferredDivisionForDefinitions": [
    "Mass.Kilogram = MassConcentration.KilogramPerCubicMeter * Volume.CubicMeter"
  ],
  "definitions": [
    "1 = Area.SquareMeter * ReciprocalArea.InverseSquareMeter",
    "1 = ElectricResistivity.OhmMeter * ElectricConductivity.SiemensPerMeter",
    "1 = Length.Meter * ReciprocalLength.InverseMeter",
    "Acceleration.MeterPerSecondSquared = Jerk.MeterPerSecondCubed * Duration.Second",
    "AmountOfSubstance.Kilomole = MolarFlow.KilomolePerSecond * Duration.Second",
    "AmountOfSubstance.Mole = Molarity.MolePerCubicMeter * Volume.CubicMeter",
    "Mass.Kilogram = MassConcentration.KilogramPerCubicMeter * Volume.CubicMeter"
  ]
}
```

Or add a structure for each definition, allowing us to configure stuff for each entry:
 
```json
{
  "definitions": [
    { "d": "1 = Area.SquareMeter * ReciprocalArea.InverseSquareMeter" },
    { "d": "1 = ElectricResistivity.OhmMeter * ElectricConductivity.SiemensPerMeter" },
    { "d": "1 = Length.Meter * ReciprocalLength.InverseMeter" },
    { "d": "Acceleration.MeterPerSecondSquared = Jerk.MeterPerSecondCubed * Duration.Second" },
    { "d": "AmountOfSubstance.Kilomole = MolarFlow.KilomolePerSecond * Duration.Second" },
    { "d": "AmountOfSubstance.Mole = Molarity.MolePerCubicMeter * Volume.CubicMeter" },
    { "d": "Mass.Kilogram = MassConcentration.KilogramPerCubicMeter * Volume.CubicMeter", "skipInferredDivision": true }
  ]
}
```

Reply:
I like option 2.2, but I also like the simple json structure we have now, so I propose a string-based solution of just appending `NoInferredDivision` to the relation, see last commits.